### ISSUE TITLE
bug of  gremlin step "has()"

### ISSUE THREAD

**Issue by choeoe** [ArcadeData/arcadedb#289]
### OS: windows wsl

Error occurred when querying "has()" on properties if the data has INFINITY field.
## Expected behavior
return vertices

## Actual behavior
java.util.concurrent.ExecutionException: org.apache.tinkerpop.gremlin.driver.exception.ResponseException: Character I is neither a decimal digit number, decimal point, nor "e" notation exponential mark.

## Steps to reproduce
Vertex alice = g.addV("person").property("hair", Double.POSITIVE_INFINITY ).next();
Vertex bob = g.addV("person").property("hair", 500 ).next();

String query =“g.V().has('hair',500.00)”;

**Comment by lvca** [comment]
Reproduced.

**Comment by lvca** [comment]
The issue is with the Gremlin conversion mechanism. If you execute `g.V().has('hair',500.00)` it works, because the comparison doesn't involve BigDecimal. For some reason, Gremlin converts in BigDecimal the value `500.00`. When it tries to compare it with the values coming from the database it fails parsing "Infinity" to build a BigDecimal.

This is the spot: org.apache.tinkerpop.gremlin.util.NumberHelper:

```java
private static BigDecimal bigDecimalValue(final Number number) {
        if (number == null) return null;
        if (number instanceof BigDecimal) return (BigDecimal) number;
        if (number instanceof BigInteger) return new BigDecimal((BigInteger) number);
        return (number instanceof Double || number instanceof Float)
                ? BigDecimal.valueOf(number.doubleValue())
                : BigDecimal.valueOf(number.longValue());
    }
```

The BigDecimal is converted from a double, but the `BigDecimal.valueOf()` method transforms it in the string "Infinity".

**Comment by lvca** [comment]
Created the PR on the Gremlin project and wrote on the Discord channel about that. Waiting for the Gremlin team.

**Comment by lvca** [comment]
This is the link to the issue in the Gremlin repository: https://issues.apache.org/jira/browse/TINKERPOP-2694?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=17480014#comment-17480014.

**Comment by choeoe** [comment]
Thank you.

**Comment by lvca** [comment]
Leaving this issue open waiting for the official TimkerPop Gremlin 3.5.3 release.

**Comment by lvca** [comment]
Fixed in Gremlin 3.5.3+
